criu/files-reg.c

```diff
@@ -2054,8 +2271,21 @@ static bool validate_file(const int fd, const struct stat *fd_status,
 		return false;
 	}
 
-	if (opts.file_validation_method == FILE_VALIDATION_BUILD_ID)
+	if (opts.file_validation_method == FILE_VALIDATION_BUILD_ID) {
 		result = validate_with_build_id(fd, fd_status, rfi);
+		if (!result) {
+			pr_info("Could not validate with build-ID, trying checksum validation instead for file %s\n",
+				rfi->path);
+			result = validate_with_checksum(fd, fd_status, rfi);
+		}
+	} else if (opts.file_validation_method == FILE_VALIDATION_CHKSM) {
+		result = validate_with_checksum(fd, fd_status, rfi);
+		if (!result) {
+			pr_info("Could not validate with checksum, trying build-ID validation instead for file %s\n",
+				rfi->path);
+			result = validate_with_build_id(fd, fd_status, rfi);
+		}
```

Comment:
I don't see how failed validation with checksum can success with build-id.

Reply:
Yeah, you're right. I've removed build-id as a fallback for checksum.